### PR TITLE
BOLT 7: add chain_hash to channel_update and channel_announcement

### PR DESCRIPTION
**Pull request by Roasbeef**
This commit adds a 32-byte `chain_hash` value to both the
`channel_update` and `channel_announcement` messages. The rationale for
this change is that this value is already present within the
`open_channel` for identifying _which_ chain to open the channel
within. As is now, if a pair of peers had channels open on two chains
which somehow are encoded using the same `short_channel_id`, then the
announcements would be ambitious. We resolve this by explicitly
including the `chain_hash` is all channel related announcement
messages.

Note that with this change, we now require 40-bytes to uniquely
identify a channel globally.

Additionally, this modification of the channel announcement messages
allows peers to start building up a heterogenous network graph.

### CODE REVIEW

File: 07-routing-gossip.md [hunk 1]
```diff
@@ -300,6 +310,11 @@ The following table specifies the meaning of the individual bits:
 The creating node MUST set `signature` to the signature of the
 double-SHA256 of the entire remaining packet after `signature` using its own `node_id`.
 
+The creating node MUST set `chain_hash` to match 32-byte hash that
+uniquely identifies the channel within the `channel_announcement` message.
```
Comment (cdecker) [inline 07-routing-gossip.md]: I guess this should be the tuple (`chain_hash`, `short_channel_id`) which uniquely identifies the channel.

File: 07-routing-gossip.md [hunk 2]
```diff
@@ -300,6 +310,11 @@ The following table specifies the meaning of the individual bits:
...
+For the Bitcoin blockchain, the `chain_hash` value MUST be (encoded in
+hex): `000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f`.
```
Comment (cdecker) [inline 07-routing-gossip.md]: We should specify the hash once and reference it thereon.

Comment (Roasbeef) [inline 07-routing-gossip.md]: Good point. Where does everyone think this definition should go? In the introduction, perhaps? 

I ended up sticking it in the glossary, but there may be a better spot for it.

File: 07-routing-gossip.md (hunk 1 above)
Comment (Roasbeef) [inline 07-routing-gossip.md]: Modified to point out that channels are uniquely identified by that two tuple.

File: 00-introduction.md [hunk 3]
```diff
@@ -110,6 +110,17 @@ This is version 0.
...
      that the feature MAY be disregarded by the other endpoint.
 
+
+* `chain_hash`:
+   * Used in several of the BOLT documents, and denotes the genesis hash of a
+     target blockchain. This allows nodes to create and reference channels on
+     several blockchains. Nodes are to ignore any messages which reference a
+     `chain_hash` that are unknown to them.
+
+     For the mahin chain Bitcoin blockchain, the `chain_hash` value MUST be
```
Comment (pm47) [inline 00-introduction.md]: typo mahin -> main

**Comment by rustyrussell** [comment]
Pierre-Marie Padiou <notifications@github.com> writes:
> +* `chain_hash`:
> +   * Used in several of the BOLT documents, and denotes the genesis hash of a
> +     target blockchain. This allows nodes to create and reference channels on
> +     several blockchains. Nodes are to ignore any messages which reference a
> +     `chain_hash` that are unknown to them.
> +
> +     For the mahin chain Bitcoin blockchain, the `chain_hash` value MUST be
>
> typo mahin -> main

He's obviously using the American pronunciation :)

**Pull request merged**